src/dmd/dcast.d

```diff
@@ -839,6 +839,27 @@ MATCH implicitConvTo(Expression e, Type t)
             }
         }
 
+        override void visit(CatExp e)
+        {
+            enum LOG = false;
+            static if (LOG)
+            {
+                printf("CatExp::implicitConvTo(this=%s, type=%s, t=%s)\n", e.toChars(), e.type.toChars(), t.toChars());
+            }
+
+            visit(cast(Expression)e);
+            if (result != MATCH.nomatch)
+                return;
+
+            if (true) // TODO: iff _all_ indirections of common element type of `e` operands are immutable
```

Comment:
Need help filling in the correct predicate here.

Reply:
Is

```d
if (!hasAliasing!(E))
{
    // allow conversion from E[] to immutable(E)[]
}
```

the correct thing here? If so just need to put together hasAliasing if it's not already defined in dmd under a different name...

Reply:
Solved.